Docs/Utilities.md

```diff
@@ -27,61 +27,98 @@ The IFreezable interface describes a freezable class, the FreezableBase abstract
-	{
-	    // If the event is triggered the entry for the cache key will be removed. It will work otherwise too but it's better to call Monitor() only when the cache entry is created newly.
-	    _cacheService.Monitor("event key", "cache key");
-	
-	    // ...
-	});
-	
-	// Removes all cache entries that were subscribed to the specified event.
-	_cacheService.Trigger("event key");
-	// You can use the above to invalidate multiple cache entries at once: when they Monitor the same event key, triggering the event will remove all of the cache entries.
+Here, we show usage examples for most of the extension methods.
+
+#### CacheServiceMonitor
+```csharp
+// _cacheService is an injected Orchard.Caching ICacheService.
+var value = _cacheService.Get<MyClass>("key", () => 
+{
+	// If the event is triggered the entry for the cache key will be removed. It will work otherwise too, but it's
+	// better to call Monitor() only when the cache entry is created newly.
+	_cacheService.Monitor("event key", "cache key");
+
+	// ...
+});
+
+// Removes all cache entries that were subscribed to the specified event.
+_cacheService.Trigger("event key");
+// You can use the above to invalidate multiple cache entries at once: when they Monitor the same event key, triggering
+// the event will remove all of the cache entries.
+```
 	
-	// HqlQueryExtensions
-	// From a Projector IFilterProvider.
-	public void ApplyFilter(FilterContext context)
+#### HqlQueryExtensions
+```csharp
+// From a Projector IFilterProvider.
+public void ApplyFilter(FilterContext context)
+{
+	IEnumerable<int> ids = ...
+	// Applies the IN() in partitions of given size so the maximal number of IN() arguments isn't an issue.
+	context.Query.WhereIdIn(ids);
+}
+```
+
+#### HqlExpressionFactoryExtensions
+... contains extensions to use such partitioned clauses in a generic way.
+
+#### ShapeDisplayingContextExtensions
+```csharp
+public class ShapeHider : ShapeDisplayEvents
+{
+	public override void Displaying(ShapeDisplayingContext context)
 	{
-	    IEnumerable<int> ids = ...
-	    context.Query.WhereIdIn(ids); // Applies the IN() in partitions of given size so the maximal number of IN() arguments isn't an issue.
+		// ...
+		context.HideShape();
 	}
-	// HqlExpressionFactoryExtensions has an extension to use such partitioned clauses in a generic way.
-
-	// ShapeDisplayingContextExtensions
-	public class ShapeHider : ShapeDisplayEvents
-    {
-        public override void Displaying(ShapeDisplayingContext context)
-        {
-			// ...
-            context.HideShape();
-        }
-    }
-
-    // StorageProviderExtensions
-    // Getting the directory separator used by an injected IStorageProvider:
-    var directorySeparator = _storageProvider.GetDirectorySeparator();
+}
+```
+
+#### StorageProviderExtensions
+```csharp
+// Getting the directory separator used by an injected IStorageProvider:
+var directorySeparator = _storageProvider.GetDirectorySeparator();
+```
+
+#### StringExtensions
+```razor

```

Comment:
According to [this file here](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L2356), `razor` is a supported language 😀